client/shipping-label/views/purchase/index.js

```diff
@@ -43,6 +53,22 @@ const PrintLabelDialog = ( props ) => {
 		return props.labelActions.purchaseLabel;
 	};
 
+	const buttons = [
+		{
+			isDisabled: ! props.form.needsPrintConfirmation && ( ! props.canPurchase || props.form.isSubmitting ),
+			onClick: getPurchaseButtonAction(),
+			isPrimary: true,
+			label: getPurchaseButtonLabel(),
+		},
+	];
+
+	if ( ! props.form.needsPrintConfirmation ) {
```

Comment:
Note that, even without the `Cancel` button, the dialog can still be closed by clicking outside it or pressing `Esc`. If we want to prevent that, you'll have to add the `needsPrintConfirmation` condition here: https://github.com/Automattic/woocommerce-services/blob/master/client/shipping-label/state/reducer.js#L64